src/darwin/Framework/CHIP/MTRDeviceControllerDelegate.h

```diff
@@ -19,25 +19,26 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
-typedef NS_ENUM(NSUInteger, MTRPairingStatus) {
-    MTRPairingStatusUnknown = 0,
-    MTRPairingStatusSuccess = 1,
-    MTRPairingStatusFailed = 2,
-    MTRPairingStatusDiscoveringMoreDevices = 3
-};
+typedef NS_ENUM(NSUInteger, MTRCommissioningStatus) {
+    MTRCommissioningStatusUnknown = 0,
+    MTRCommissioningStatusSuccess = 1,
+    MTRCommissioningStatusFailed = 2,
+    MTRCommissioningStatusDiscoveringMoreDevices = 3
+} MTR_NEWLY_AVAILABLE;
 
 /**
- * The protocol definition for the MTRDevicePairingDelegate
+ * The protocol definition for the MTRDeviceControllerDelegate
  *
  * All delegate methods will be called on the supplied Delegate Queue.
  */
-@protocol MTRDevicePairingDelegate <NSObject>
+MTR_NEWLY_AVAILABLE
+@protocol MTRDeviceControllerDelegate <NSObject>
 @optional
 /**
  * Notify the delegate when pairing status gets updated
  *
  */
-- (void)onStatusUpdate:(MTRPairingStatus)status;
+- (void)onStatusUpdate:(MTRCommissioningStatus)status;
```

Comment:
If we're changing the API anyway, we should consider making this more like the normal delegate pattern and pass the `MTRDeviceController` the call is coming from, e.g. `deviceController:didCompletePairing:`. Generally with the delegate pattern the assumption is that you could use the same delegate for multiple objects, e.g. if you had multiple fabrics with their own `MTRDeviceController` you could still point them at the same delegate and make sense of the events.

I'm also wondering if some of these callbacks should provide more information, e.g. `OnPairingDeleted:` doesn't seem to provide any information about which pairing was deleted?

Reply:
> we should consider making this more like the normal delegate pattern and pass the MTRDeviceController the call is coming from

Coming up when we reland PR #22682.

> e.g. OnPairingDeleted: doesn't seem to provide any information about which pairing was deleted?

It's getting removed when we reland PR #22676.

But yes, some of these callbacks should provide more information in various ways; the problem is that the underlying SDK does not do that.  So it's more than just a Darwin change too fix that.  But we do in fact want to address that in the next few months....